trace/trace.go

```diff
@@ -206,6 +206,21 @@ func startSpanInternal(name string, hasParent bool, parent SpanContext, remotePa
 	span.spanContext = parent
 
 	cfg := config.Load().(*Config)
+	gen := cfg.IDGenerator.(*defaultIDGenerator)
+
+	if !gen.randomIDsGenerated {
```

Comment:
It is possible to start multiple spans in parallel (in separate goroutines), so I think we need to introduce a lock here to make sure we initialize the generator exactly once?

I also suggest to extract this code into a function belonging to `defaultIDGenerator` and just invoke it from here, e.g.

```go
gen, ok := cfg.IDGenerator.(*defaultIDGenerator)
if ok {
  gen.initialize()
}
```
```go

type defaultIDGenerator struct {
  ...
  initOnce sync.Once
  ...
}

// initOnce initializes the generator on the first call
func (gen *defaultIDGenerator) initialize() {
  gen.initOnce.Do(func(){
    // initialize traceID and spanID generators.
    ...
  })
}
```

Reply:
Thanks, `sync.Once` fits nicely here. Updated with that!

(I wanted to give you `Co-authored-by` on this, but couldn't find an email address to use. Feel free to edit the commit message and add yourself if you'd like.)